File: Beginner/265/b.py
n, m, t = map(int, input().split())
a_n = list(map(int, input().split()))
xy_m = [list(map(int, input().split())) for _ in range(m)]

if m == 0:
    if t <= sum(a_n):
        print("No")
    else:
        print("Yes")
    exit()

current_time = t
bounus_idx = 0
for i in range(len(a_n)):
    if xy_m[bounus_idx][0] - 1 == i:
        current_time += xy_m[bounus_idx][1]
        if len(xy_m) - 1 > bounus_idx:
            bounus_idx += 1
    current_time -= a_n[i]
    if current_time <= 0:
        print("No")
        exit()
print("Yes")

File: Beginner/265/b.go
package main

import "fmt"

func main() {
	var n, m, t int
	fmt.Scan(&n, &m, &t)
	a_n := make([]int, n-1)
	for i := 0; i < n-1; i++ {
		fmt.Scan(&a_n[i])
	}
	xy_m := make([][2]int, m)
	for i := 0; i < m; i++ {
		fmt.Scan(&xy_m[i][0], &xy_m[i][1])
	}
	if m == 0 {
		sum_a_n := 0
		for i := 0; i < n-1; i++ {
			sum_a_n += a_n[i]
		}
		if sum_a_n >= t {
			fmt.Println("No")
			return
		} else {
			fmt.Println("Yes")
			return
		}
	}
	var currentTime = t
	var bounusIdx = 0
	for i := 0; i < n-1; i++ {
		if xy_m[bounusIdx][0]-1 == i {
			currentTime += xy_m[bounusIdx][1]
			if bounusIdx < m-1 {
				bounusIdx++
			}
		}
		currentTime -= a_n[i]
		if currentTime <= 0 {
			fmt.Println("No")
			return
		}
	}
	fmt.Println("Yes")
}
